fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * FDBDirectoryLockFactory.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.directory;
+
+import org.apache.lucene.store.AlreadyClosedException;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.Lock;
+import org.apache.lucene.store.LockFactory;
+
+import java.io.IOException;
+
+/**
+ * Produce a lock over {@link FDBDirectory}.
+ */
+public final class FDBDirectoryLockFactory extends LockFactory {
+    final FDBDirectory directory;
+
+    public FDBDirectoryLockFactory(FDBDirectory directory) {
+        this.directory = directory;
+    }
+
+    @Override
+    public Lock obtainLock(final Directory dir, final String lockName) throws IOException {
+        // dir is ignored
```

Comment:
I can simply make the `dir`'s path a part of the `lockName` as our lock key.  Either as a String or as hash. Would that be sufficient? 
(My conjecture was that it's safer to trust the FDBDirectory than the given "dir" - I was not aware that they may be different)

Reply:
ok, yeah, I checked the code, and it doesn't ever actually pass anything other than `this`. It might make sense to remove the field, and unwrap the given directory, but it probably doesn't matter.